src/components/DocExplorer.js

```diff
@@ -148,6 +181,142 @@ export class DocExplorer extends React.Component {
   }
 }
 
+// Render Search Results
+class SearchDoc extends React.Component {
+  shouldComponentUpdate(nextProps) {
+    return (this.props.schema !== nextProps.schema) ||
+      (this.props.searchValue !== nextProps.searchValue);
+  }
+
+  render() {
+    const searchValue = this.props.searchValue;
+    const schema = this.props.schema;
+    const onClickType = this.props.onClickType;
+    const onClickField = this.props.onClickField;
+
+    const typeMap = schema.getTypeMap();
+
+    // Compose a search index for all possible arguments from every types.
+    let possibleArguments = [];
+    Object.keys(typeMap).forEach(typeName => {
+      let type = typeMap[typeName];
+      if (type.getFields) {
+        let fields = type.getFields();
+        Object.keys(fields).forEach(fieldName => {
+          let field = fields[fieldName];
+          if (field.args && field.args.length) {
+            Array.prototype.push.apply(possibleArguments, field.args);
+          }
+        });
+      }
+    });
+
+    let matchedTypes = [];
+    let matchedFields = [];
+    let matchedArgs = [];
+
+    Object.keys(typeMap).forEach(typeName => {
+      const type = typeMap[typeName];
+      let matchedOn = [];
+      if (typeName.indexOf(searchValue) !== -1) {
+        matchedOn.push('Type Name');
+      }
+
+      if (type.description && type.description.indexOf(searchValue) !== -1) {
```

Comment:
I would remove this for this initial PR - I'm concerned that searching over descriptions will be expensive for larger schema, and perhaps we'll want to do a specialized index for this. Let's come back to it in a separate PR once the main search functionality is working.